### PR TITLE
Fix WFT failures sometimes getting stuck in a spam loop

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
While working on LA implementation, discovered that all-internal state machine errors would not avoid spamming WFT failures properly. Fixed that.

## Why?
:bug: 

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: src/worker/mod.rs [hunk 1]
```diff
@@ -538,7 +546,7 @@ impl Worker {
         &self,
         run_id: &str,
         success: workflow_completion::Success,
-    ) -> Result<bool, CompleteWfError> {
+    ) -> Result<WFTReportOutcome, CompleteWfError> {
```
Comment (cretz) [inline src/worker/mod.rs:549]: Pedantic - Cannot tell with `Wf` vs `WFT` whether we should capitalize abbreviations or not

**Review by cretz — APPROVED** [review]

File: src/worker/mod.rs (hunk 1 above)
Comment (Sushisource) [reply src/worker/mod.rs:549]: Yeah I've been kinda inconsistent here. Clippy by default says don't, but I kinda like it. :man_shrugging:

File: src/errors.rs [hunk 2]
```diff
@@ -13,16 +13,13 @@ pub(crate) struct WorkflowUpdateError {
...
-    /// The task token associated with this update, if one existed yet.
-    pub task_token: Option<TaskToken>,
 }
 
 impl From<WorkflowMissingError> for WorkflowUpdateError {
     fn from(wme: WorkflowMissingError) -> Self {
         Self {
             source: WFMachinesError::Fatal("Workflow machines missing".to_string()),
             run_id: wme.run_id,
-            task_token: None,
```
Comment (bergundy) [inline src/errors.rs:25]: Why did you remove this?

Comment (Sushisource) [reply src/errors.rs:25]: It's no longer needed for anything.

File: src/worker/mod.rs [hunk 3]
```diff
@@ -648,35 +648,47 @@ impl Worker {
     async fn wf_activation_failed(
         &self,
         run_id: &str,
+        cause: WorkflowTaskFailedCause,
         failure: workflow_completion::Failure,
-    ) -> Result<bool, CompleteWfError> {
+    ) -> Result<WFTReportOutcome, CompleteWfError> {
+        warn!(run_id, failure=?failure, "Failing workflow activation");
```
[the remark this answers is not in the document]
Comment (Sushisource) [reply src/worker/mod.rs:654]: I suppose if something is going wrong it could be... but Shawn did specifically mention users wanting to see this in the logs. I could shift it to be only when we *report* to server which would make sense. I'll do that.
